internal/impl/postgresql/integration_test.go

```diff
@@ -0,0 +1,939 @@
+//
+// https://github.com/redpanda-data/connect/v4/blob/main/licenses/rcl.md
+
+package pgstream
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/jaswdr/faker"
+	_ "github.com/lib/pq"
+	_ "github.com/redpanda-data/benthos/v4/public/components/io"
+	_ "github.com/redpanda-data/benthos/v4/public/components/pure"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+)
+
+func ResourceWithPostgreSQLVersion(t *testing.T, pool *dockertest.Pool, version string) (*dockertest.Resource, *sql.DB, error) {
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "postgres",
+		Tag:        version,
+		Env: []string{
+			"POSTGRES_PASSWORD=secret",
+			"POSTGRES_USER=user_name",
+			"POSTGRES_DB=dbname",
+		},
+		Cmd: []string{
+			"postgres",
+			"-c", "wal_level=logical",
+		},
+	}, func(config *docker.HostConfig) {
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{Name: "no"}
+	})
+
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, pool.Purge(resource))
+	})
+
+	require.NoError(t, resource.Expire(120))
+
+	hostAndPort := resource.GetHostPort("5432/tcp")
+	hostAndPortSplited := strings.Split(hostAndPort, ":")
+	databaseURL := fmt.Sprintf("user=user_name password=secret dbname=dbname sslmode=disable host=%s port=%s", hostAndPortSplited[0], hostAndPortSplited[1])
+
+	var db *sql.DB
+	pool.MaxWait = 120 * time.Second
+	if err = pool.Retry(func() error {
+		if db, err = sql.Open("postgres", databaseURL); err != nil {
+			return err
+		}
+
+		if err = db.Ping(); err != nil {
+			return err
+		}
+
+		var walLevel string
+		if err = db.QueryRow("SHOW wal_level").Scan(&walLevel); err != nil {
+			return err
+		}
+
+		var pgConfig string
+		if err = db.QueryRow("SHOW config_file").Scan(&pgConfig); err != nil {
+			return err
+		}
+
+		if walLevel != "logical" {
+			return fmt.Errorf("wal_level is not logical")
+		}
+
+		_, err = db.Exec("CREATE TABLE IF NOT EXISTS flights (id serial PRIMARY KEY, name VARCHAR(50), created_at TIMESTAMP);")
+		if err != nil {
+			return err
+		}
+
+		// flights_non_streamed is a control table with data that should not be streamed or queried by snapshot streaming
+		_, err = db.Exec("CREATE TABLE IF NOT EXISTS flights_non_streamed (id serial PRIMARY KEY, name VARCHAR(50), created_at TIMESTAMP);")
+
+		return err
+	}); err != nil {
+		panic(fmt.Errorf("could not connect to docker: %w", err))
+	}
+
+	return resource, db, nil
+}
+
+func TestIntegrationPgCDC(t *testing.T) {
+	tmpDir := t.TempDir()
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	// Use custom PostgreSQL image with wal2json plugin compiled in
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "usedatabrew/pgwal2json",
```

Comment:
Can we use https://hub.docker.com/r/debezium/postgres/dockerfile instead? I'm assuming you're not going to maintain writing new images as new version of postgres come out.